### PR TITLE
Fix TLS connection of the pure Ruby reactor

### PR DESCRIPTION
**Pull request by meineerde**
With these commits, I was able to successfully run `rake test_pure` without any failing tests.

### CODE REVIEW

**Comment by sodabrew** [comment]
Thank you! Running CI now

**Comment by sodabrew** [comment]
Actually, it occurs to me that the GitHub CI test suite does not have a pure-ruby mode. Would you be up for adding it at least one row to the matrix with one version of Ruby that runs it?

**Comment by meineerde** [comment]
I added a test step. Unfortunately, some tests do still fail (on my Mac) when running `rake test_em_pure_ruby`. My previous changes improve the situation compared to what was before, but apparently there is still some differing behavior...

It might still be necessary to mark the build test as not required (i.e. something like `allow_failures` on Travis CI), but I'm not 100% sure how to correctly configure this with Github Actions.

File: lib/em/pure_ruby.rb [hunk 1]
```diff
@@ -280,7 +280,7 @@ def set_tls_parms signature, priv_key_path, priv_key, priv_key_pass, cert_chain_
       @tls_parms[signature][:priv_key] = priv_key if tls_parm_set?(priv_key)
       @tls_parms[signature][:priv_key_pass] = priv_key_pass if tls_parm_set?(priv_key_pass)
       @tls_parms[signature][:cert_chain] = File.read(cert_chain_path) if tls_parm_set?(cert_chain_path)
-      @tls_parms[signature][:cert_chain] = cert if tls_parm_set?(cert_chain)
+      @tls_parms[signature][:cert_chain] = cert if tls_parm_set?(cert)
```
Comment (sodabrew) [inline lib/em/pure_ruby.rb:283]: I'm looking if this was a bug previously? cert vs. cert_chain in the if block

**Comment by sodabrew** [comment]
I posted a question here, and otherwise this PR looks subsumed into #993 cc @nevans -- Thank you both!!

**Pull request merged**